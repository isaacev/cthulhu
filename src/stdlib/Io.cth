intrinsic {
  fn php_print(Str) -> ();
  fn read_argv() -> [Str];
}

#[inline]
pub fn print(str: Str) -> () {
  php_print(str)
}

#[inline]
pub fn println(str: Str) -> () {
  php_print(str ++ "\n")
}

pub fn argv() -> [Str] {
  read_argv()
}
