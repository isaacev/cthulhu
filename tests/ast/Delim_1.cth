use ::Io;

pub type {
  alias Token =
    { kind: Token::Kind
    , lexeme: Str
    }
}

blah blah blah
