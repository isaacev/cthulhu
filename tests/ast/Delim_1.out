
  EXPECTED TOKEN  ------------------------------------------------------------

    : Delim_1
  3 | pub type {
    |          ^ expected uppercase identifier here

